hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -233,7 +233,7 @@ void doRemoveAddress(Address deadAddress, boolean destroyConnection) {
             }
             Connection conn = node.connectionManager.getConnection(deadAddress);
             if (destroyConnection && conn != null) {
-                node.connectionManager.destroyConnection(conn);
+                conn.close(reason, null);
```

Comment:
`ConnectionManager.destroyConnection()` has more logic than `Connection.close()`. I'm not sure if this is a valid optimization.